zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -394,6 +409,7 @@ private void broadcast(String noteId, Message m) {
         return;
       }
       LOG.debug("SEND >> " + m.op);
+      broadcastToWatchers(noteId, StringUtils.EMPTY, m);
```

Comment:
We can move this one before if statement above to make sure message is broadcasted even if no client connection opened note.